aquadoggo/src/db/models/log.rs

```diff
@@ -120,43 +120,48 @@ impl Log {
 
     /// Returns registered or possible log id for a document.
     ///
-    /// If no log has been previously registered for this document automatically returns the next
-    /// unused log_id.
-    /// SYSTEM schema log ids are pre-defined by the protocol specification.
+    /// If no log has been previously registered for this document it automatically returns the
+    /// next unused log_id.
     pub async fn find_document_log_id(
         pool: &Pool,
         author: &Author,
-        document: &Hash,
+        document_hash: Option<&Hash>,
     ) -> Result<LogId> {
-        // Determine log_id for this document
-        let document_log_id = Log::get(pool, author, document).await?;
+        // Determine log_id for this document when a hash was given
+        let document_log_id = match document_hash {
+            Some(hash) => Log::get(pool, author, hash).await?,
+            None => None,
+        };
 
-        // Use result or find next possible log_id automatically
+        // Use result or find next possible log_id automatically when nothing was found yet
         let log_id = match document_log_id {
             Some(value) => value,
-            None => Log::next_user_schema_log_id(pool, author).await?,
+            None => Log::next_log_id(pool, author).await?,
         };
 
         Ok(log_id)
     }
 
-    /// Returns the registered log id for an instance.
+    /// Returns the related document for any entry.
     ///
-    /// Every instance is part of a document and, through that, associated with a specific log id
-    /// of its author. This method returns that log id by looking up the log that the instance's
-    /// last operation was stored in.
-    pub async fn get_log_id_by_instance(pool: &Pool, instance: &Hash) -> Result<Option<LogId>> {
-        let result = query_as::<_, LogId>(
+    /// Every entry is part of a document and, through that, associated with a specific log id used
+    /// by this document and author. This method returns that document by looking up the log id
+    /// that the document's last operation was stored in.
```

Comment:
Not sure if I understand this:

> looking up the log id that the document's last operation was stored in.

Where does the "last operation" come into this? As I understand this query, we're looking up the `document_id` from the log where the passed entry is stored.

Reply:
Yes, thats confusing! I'll remove it :+1: